pyexasol/http_transport.py

```diff
@@ -6,6 +9,245 @@
+            ip_address:port/sha256_base64_encoded_string
+        """
+        pattern = r"\/([a-zA-Z0-9_\-+\/]+=)"
+        match = re.search(pattern, exa_address)
+        if match:
+            return match.group(1)
+        return None
+
+    @staticmethod
+    def _get_query_str(query_lines: list[Optional[str]]) -> str:
+        filtered_query_lines = [
+            query_line for query_line in query_lines if query_line is not None
+        ]
+        return "\n".join(filtered_query_lines)
+
+    def _requires_tls_public_key(self) -> bool:
+        if (
+            exasol_db_version := self.connection.exasol_db_version
+        ) and exasol_db_version >= Version("8.32.0"):
+            if self.connection.options["encryption"]:
+                return True
+        return False
+
+    @property
+    def _column_delimiter(self) -> Optional[str]:
+        if self.column_delimiter:
+            return f"COLUMN DELIMITER = {self.connection.format.quote(self.column_delimiter)}"
+        return None
+
+    @property
+    def _column_separator(self) -> Optional[str]:
+        if self.column_separator:
+            return f"COLUMN SEPARATOR = {self.connection.format.quote(self.column_separator)}"
+        return None
+
+    @property
+    def _comment(self) -> Optional[str]:
+        if self.comment:
+            if "*/" in self.comment:
+                raise ValueError("Invalid comment, cannot contain */")
+            return f"/*{self.comment}*/"
+        return None
+
+    @property
+    def _encoding(self) -> Optional[str]:
+        if self.encoding:
+            return f"ENCODING = {self.connection.format.quote(self.encoding)}"
+        return None
+
+    @property
+    def _null(self) -> Optional[str]:
+        if self.null:
+            return f"NULL = {self.connection.format.quote(self.null)}"
+        return None
+
+    @property
+    def _row_separator(self) -> Optional[str]:
+        if self.row_separator:
+            return f"ROW SEPARATOR = {self.connection.format.quote(self.row_separator)}"
+        return None
+
+
+@dataclass
+class ImportQuery(SqlQuery):
+    # set these values in param dictionary to ExaConnection
+    skip: Optional[str] = None
+    trim: Optional[str] = None
+
+    def build_query(self, source: str, exa_address_list: list[str]) -> str:
+        query_lines = [
+            self._comment,
+            self._get_import(source=source),
+            *self._get_file_list(exa_address_list=exa_address_list),
+            self._encoding,
+            self._null,
+            self._skip,
+            self._trim,
+            self._row_separator,
+            self._column_separator,
+            self._column_delimiter,
+        ]
+        return self._get_query_str(query_lines)
+
+    @staticmethod
+    def load_from_dict(
+        connection: ExaConnection, compression: bool, params: dict
+    ) -> ImportQuery:
+        return ImportQuery(connection=connection, compression=compression, **params)
```

Comment:
note that if folks pass in other values to params than expected that the behavior has changed -> before we just pulled values we expected from the params dict and now we pass them into a class, i.e.. they have "dummy", it will now raise an exception saying "dummy" isn't a valid value